source/h1_connection.c

```diff
@@ -546,6 +546,34 @@ static void s_stream_complete(struct aws_h1_stream *stream, int error_code) {
         }
     }
 
+    if (error_code != AWS_ERROR_SUCCESS) {
+        if (stream->base.client_data && stream->is_incoming_message_done && stream->is_outgoing_message_done == false) {
+            /* As a request that finished receiving the response, we ignore error and
+             * consider it finished successfully */
+            AWS_LOGF_DEBUG(
+                AWS_LS_HTTP_STREAM,
+                "id=%p: Request stream haven't finished sending, but error code %d (%s) happened and full response "
+                "received. Stop sending and treated as succeed",
+                (void *)&stream->base,
+                error_code,
+                aws_error_name(error_code));
+            error_code = AWS_ERROR_SUCCESS;
+        }
+        if (stream->base.server_data && stream->is_outgoing_message_done && stream->is_incoming_message_done == false) {
+            /* As a server finished sending the response, but still failed with the request was not finished receiving.
+             * We ignore error and consider it finished successfully
+             */
+            AWS_LOGF_DEBUG(
+                AWS_LS_HTTP_STREAM,
+                "id=%p: Server finishes sending response, but error code %d (%s) happens before receives the full "
+                "response. Treated as succeed",
```

Comment:
similar feedback, do we need to check `is_incoming_message_done`?

and maybe

"id=%p: Ignoring error code %d (%s). The response has  been fully sent,"
" so the stream will complete successfully"